airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -155,9 +156,32 @@ def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[
         params.update(**stream_slice)
         return params
 
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            if self.cursor_field in record:
+                record[self.cursor_field] = pendulum.from_timestamp(float(record[self.cursor_field]))
+
+            yield record
+
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
         current_stream_state = current_stream_state or {}
-        current_stream_state["ts"] = max(float(latest_record["ts"]), current_stream_state.get("ts", 0))
+        if isinstance(latest_record[self.cursor_field], DateTime):
+            latest_record[self.cursor_field] = latest_record[self.cursor_field].timestamp()
```

Comment:
wow, this part is super smell, this is `get_*` method and we are modifying latest_record here, also it is annotated as read only Mapping

Reply:
Can it be interpreted as getting updated state? Changed latest_record to MutableMapping and added return annotation

Reply:
I mean we not suppose to change records here, what is the reason for doing so?
why can't we do this in `parse_response` or `read_records`?

Reply:
we are not change records here anymore